### PR TITLE
PLT-8510: Feature checks for older webkit support of Performance API.

### PR DESCRIPTION
**Pull request by mkraft**
#### Summary
Adds feature checks for Performance API support.

#### Ticket Link
[PLT-8510](https://mattermost.atlassian.net/browse/PLT-8510)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed

### CODE REVIEW

File: actions/diagnostics_actions.jsx [hunk 1]
```diff
@@ -77,7 +84,37 @@ export function measure(name1, name2) {
...
+        trackEvent('performance', 'page_load', {duration: pageLoadTime});
+    }, tenSeconds);
+}
+
 function mostRecentDurationByEntryName(entryName) {
     const entriesWithName = performance.getEntriesByName(entryName);
     return entriesWithName.map((item) => item.duration)[entriesWithName.length - 1];
+}
+
+function isSupported(checks) {
```
Comment (hmhealey) [inline actions/diagnostics_actions.jsx]: Maybe we could just do this once and store the results? That might help reduce the impact of these checks

File: actions/diagnostics_actions.jsx [hunk 2]
```diff
@@ -57,7 +57,14 @@ export function mark(name) {
  *
  */
 export function measure(name1, name2) {
-    if (!global.mm_config.EnableDeveloper === 'true') {
+    const performanceMethodsUsed = [
```
Comment (hmhealey) [inline actions/diagnostics_actions.jsx]: Do we need to check for the individual functions, or is it sufficient to check if `window.performance` is set?

Comment (mkraft) [reply actions/diagnostics_actions.jsx]: Not sure if we *need* to. MDN docs do this and the compatibility tables aren't up-to-date so I can't say for sure without a trying on various machines. I generally don't really worry about performance in development mode.

File: actions/diagnostics_actions.jsx (hunk 1 above)
Comment (mkraft) [reply actions/diagnostics_actions.jsx]: In redux?

Comment (mkraft) [reply actions/diagnostics_actions.jsx]: Done.

Comment (mkraft) [reply actions/diagnostics_actions.jsx]: Done.

**Comment by jasonblais** [comment]
@mkraft Can you help rebase against the 4.6 branch?

**Comment by mkraft** [comment]
@jasonblais Yes will do.

**Comment by mkraft** [comment]
@jasonblais Should be good now.

**Comment by jasonblais** [comment]
Thanks @mkraft! Looks like the build is still failing though.

**Comment by jwilander** [comment]
Seeing a ton of these:

```
FAIL tests/components/integrations/add_command.test.jsx
  ● Test suite failed to run

    ReferenceError: performance is not defined
```

in the build output

**Comment by mkraft** [comment]
Added a `performance` object to the `global` object for tests.

**Pull request merged**